### PR TITLE
LibPDF: Add support for the LZW Filter

### PR DESCRIPTION
**Pull request by LucasChollet**
@nico

### CODE REVIEW

File: Userland/Libraries/LibPDF/Filter.cpp [hunk 1]
```diff
@@ -206,16 +206,10 @@ PDFErrorOr<ByteBuffer> Filter::decode_png_prediction(Bytes bytes, int bytes_per_
     return decoded;
 }
 
-PDFErrorOr<ByteBuffer> Filter::decode_lzw(ReadonlyBytes)
+PDFErrorOr<ByteBuffer> Filter::handle_lzw_and_flate_parameters(ByteBuffer buffer, int predictor, int columns, int colors, int bits_per_component)
 {
```
Comment (nico) [inline Userland/Libraries/LibPDF/Filter.cpp:215]: Consider adding `// Table 3.7 Optional parameters for LZWDecode and FlateDecode filters` as first line.

File: Userland/Libraries/LibPDF/Filter.cpp [hunk 2]
```diff
@@ -225,9 +230,32 @@ PDFErrorOr<ByteBuffer> Filter::handle_lzw_and_flate_parameters(ByteBuffer buffer
...
+
+        if (code == clear_code) {
+            lzw_decoder.reset();
+            continue;
+        }
+
+        if (code == end_of_data_code)
+            break;
+
+        TRY(decoded.try_append(lzw_decoder.get_output()));
```
Comment (nico) [inline Userland/Libraries/LibPDF/Filter.cpp:255]: This loop is the same as in LibTiff, right? Should there be a `LZWDecoder::decompress_all()` that has this loop? (Similar to DeflateDecompressor.)

**Review by nico — APPROVED** [review]
Very cool!

**Review by nico — APPROVED** [review]
(meant to hit 'approve')

File: Userland/Libraries/LibPDF/Filter.cpp (hunk 2 above)
Comment (LucasChollet) [reply Userland/Libraries/LibPDF/Filter.cpp:255]: If I understand correctly, control codes are not specified by the original LZW paper, but all user (that I know) use the same ones.
And also, GIF expect results output code per output code, TIFF byte per byte and PDF a whole decoded stream.
So it's definitely factorizable, but let's do that later.

**Review by nico — APPROVED** [review]

**Pull request merged**